File: Placement Sheet/Arrays/Initial Startups/Matrix_Is_Symmetric.java
public class Matrix_Is_Symmetric {
    public static void main(String[] args) {
        int[][] matrix1 = {
                { 1, 2, 3 },
                { 2, 4, 5 },
                { 3, 5, 8 }
        };

        if (checkSymmetricMatrixBruteForce(matrix1)) {
            System.out.println("Matrix is symmetric");
        } else {
            System.out.println("Matric is not symmetric");
        }
    }

    // Brute-Force
    private static boolean checkSymmetricMatrixBruteForce(int[][] matrix1) {
        int[][] newMatrix = new int[matrix1[0].length][matrix1.length];
        for (int i = 0; i < matrix1[0].length; i++) {
            for (int j = 0; j < matrix1.length; j++) {
                newMatrix[i][j] = matrix1[j][i];
            }
        }
        for (int i = 0; i < matrix1[0].length; i++) {
            for (int j = 0; j < matrix1.length; j++) {
                if (matrix1[i][j] != newMatrix[i][j]) {
                    return false;
                }
            }
        }
        return true;
    }

    // Optimal
    private static boolean checkSymmetricMatrix(int[][] matrix) {
        int n = matrix.length;
        for (int i = 0; i < n; i++) {
            if (matrix[i].length != n) {
                return false;
            }
        }

        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (matrix[i][j] != matrix[j][i]) {
                }
            }
        }
        return true;
    }
}

File: Placement Sheet/Arrays/Initial Startups/Python/Matrix_Is_Symmetric.py
def checkSymmetricMatrixBruteForce(arr):
    try:
        row = len(arr)
        column = len(arr[0])
        newMatrix = [list(row) for row in zip(*arr)]

        for i in range(row):
            for j in range(column):
                if arr[i][j] != newMatrix[i][j]:
                    return False
        return True

    except Exception as e:
        print(f"Error: {e}")

def checkSymmetricMatrix(arr):
    try:
        n = len(arr)
        # Ensure matrix is square
        for row in arr:
            if len(row) != n:
                return False
        # Compare arr[i][j] with arr[j][i]
        for i in range(n):
            for j in range(i + 1, n):
                if arr[i][j] != arr[j][i]:
                    return False
        return True
    except Exception as e:
        print(f"Error: {e}")


arr = [[1, 2, 3],
            [2, 4, 5],
                [3, 5, 8]]
print(checkSymmetricMatrixBruteForce(arr=arr))
print(checkSymmetricMatrix(arr=arr))